microprofile/openapi/src/test/java/io/helidon/microprofile/openapi/ServerModelReaderTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2019, 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.microprofile.openapi;
+
+import java.net.HttpURLConnection;
+
+import io.helidon.config.Config;
+import io.helidon.config.ConfigSources;
+import io.helidon.microprofile.openapi.test.MyModelReader;
+import io.helidon.nima.webserver.WebServer;
+import io.helidon.openapi.OpenApiFeature;
+
+import jakarta.json.JsonException;
+import jakarta.json.JsonString;
+import jakarta.json.JsonStructure;
+import jakarta.json.JsonValue;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+/**
+ * Makes sure that the app-supplied model reader participates in constructing
+ * the OpenAPI model.
+ */
+// TODO This test was transplanted from SE and needs to be adapted to MP.
+public class ServerModelReaderTest {
+
+    private static final String SIMPLE_PROPS_PATH = "/openapi";
+
+    private static final MpOpenApiFeature.Builder<?, ?> OPENAPI_SUPPORT_BUILDER =
+        MpOpenApiFeature.builder()
+                .config(Config.create(ConfigSources.classpath("simple.properties"))
+                                .get(OpenApiFeature.Builder.CONFIG_KEY));
+
+    private static WebServer webServer;
+
+    @BeforeAll
+    public static void startup() {
+//        webServer = TestUtil.startServer(OPENAPI_SUPPORT_BUILDER);
+    }
+
+    @AfterAll
+    public static void shutdown() {
+//        TestUtil.shutdownServer(webServer);
+    }
+
+//    @Test
```

Comment:
Why is the code below commented out ? Is it existing code that is moved ?

Reply:
This test used to be in `helidon/openapi` in Helidon 3.x. I'm not sure where it went in the initial move to 4.x but we should have it. So I copied it from 3.x.

It needs to be in an MP-aware module, and `helidon/openapi` was MP-aware in 3.x but not in 4.x The APIs the test uses are different now and I did not want to take the time now to rewrite the test to use the MP-friendly test infrastructure.

I can remove this file from the PR and add it back in rewritten form later in a separate PR or leave it here and come back and edit it later.

Reply:
I have removed the heavily-commented test and added #7106 to track the need to add it back.